Setup.bat

```diff
@@ -27,14 +36,49 @@ call :MarkStartOfBlock "Setup the git hooks"
 call :MarkEndOfBlock "Setup the git hooks"
 
 call :MarkStartOfBlock "Check dependencies"
-    if not defined UNREAL_HOME (
-        echo Error: Please set UNREAL_HOME environment variable to point to the Unreal Engine folder.
+
+    rem Find the Unreal Engine used to build this project.
+    set UNREAL_ENGINE=""
+
+    rem Get the Unreal Engine used by this project by querying the registry for the engine association found in the .uproject.
+    for /f "delims=" %%A in (' powershell -Command "Get-Childitem -Path %ProjectDirectory% -Recurse -Include *.uproject -File | %% {$_.FullName}" ') do set "UPROJECT=%%A"
+    echo Using uproject: %UPROJECT%
+
+    rem Get the Engine Association from the uproject.
+    for /f "delims=" %%A in (' powershell -Command "Select-String -Pattern '..{8}-.{4}-.{4}-.{4}-.{12}.' %UPROJECT% -AllMatches | %% { $_.Matches } | %% { $_.Value }" ') do set "ENGINE_ASSOCIATION=%%A"

```

Comment:
Maybe look into this: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-json?view=powershell-6 instead of regex